extension-requests/script.js

```diff
@@ -0,0 +1,221 @@
+const container = document.querySelector('.container');
+const extensionRequestsContainer = document.querySelector(
+  '.extension-requests',
+);
+
+const errorHeading = document.querySelector('h2#error');
+const modalParent = document.querySelector('.extension-requests-modal-parent');
+const closeModal = document.querySelectorAll('#close-modal');
+
+//modal containers
+const modalShowInfo = document.querySelector('.extension-requests-info');
+const modalStatusForm = document.querySelector(
+  '.extension-requests-status-form',
+);
+const modalUpdateForm = document.querySelector('.extension-requests-form');
+
+const state = {
+  currentExtensionRequest: null,
+};
+
+const render = async () => {
+  try {
+    addLoader(container);
+    const extensionRequests = await getExtensionRequests();
+    const allExtensionRequests = extensionRequests.allExtensionRequests;
+    allExtensionRequests.forEach((data) => {
+      extensionRequestsContainer.appendChild(createExtensionRequestCard(data));
+    });
+  } catch (error) {
+    errorHeading.textContent = 'Something went wrong';
+    errorHeading.classList.add('error-visible');
+    reload();
+  } finally {
+    removeLoader();
+  }
+};
+const showTaskDetails = async (taskId, approved) => {
+  if (!taskId) return;
+  try {
+    modalShowInfo.innerHTML = '<h3>Task Details</h3>';
+    addLoader(modalShowInfo);
+    const taskDetails = await getTaskDetails(taskId);
+    const taskData = taskDetails.taskData;
+    modalShowInfo.append(createTaskInfoModal(taskData, approved));
+  } catch (error) {
+    errorHeading.textContent = 'Something went wrong';
+    errorHeading.classList.add('error-visible');
+    reload();
+  } finally {
+    removeLoader();
+  }
+};
+function createTaskInfoModal(data, approved) {
+  if (!data) return;
+
+  const dataHeadings = [
```

Comment:
Create `constants.js` and export array from this file.

Reply:
i can but this is not going to get used anywhere else.

Reply:
I think we should still add this in constant file since it will be easy to modify in the constant file if we need to.